pkg/featuregate/featuregate.go

```diff
@@ -40,6 +40,13 @@ var (
 		featuregate.WithRegisterDescription("enables feature to set GOMEMLIMIT and GOMAXPROCS automatically"),
 		featuregate.WithRegisterFromVersion("v0.100.0"),
 	)
+	// EnableTargetAllocatorMTLS is the feature gate that enables mTLS between the target allocator and the collector.
+	EnableTargetAllocatorMTLS = featuregate.GlobalRegistry().MustRegister(
+		"operator.targetallocator.mtls",
+		featuregate.StageAlpha,
+		featuregate.WithRegisterDescription("enables mTLS between the target allocator and the collector"),
+		featuregate.WithRegisterFromVersion("v0.109.0"),
```

Comment:
```suggestion
		featuregate.WithRegisterFromVersion("v0.111.0"),
```
We've just released 0.110.0, so this should go up to the next version.